server/client.go

```diff
@@ -767,16 +767,23 @@ func (c *client) setPermissions(perms *Permissions) {
 
 // Check to see if we have an expiration for the user JWT via base claims.
 // FIXME(dlc) - Clear on connect with new JWT.
-func (c *client) checkExpiration(claims *jwt.ClaimsData) {
+func (c *client) setExpiration(claims *jwt.ClaimsData, validFor time.Duration) {
+	tn := time.Now().Unix()
+	expiresAt := time.Duration(0)
+	if claims.Expires > tn {
+		expiresAt = time.Duration(claims.Expires - tn)
+	}
 	if claims.Expires == 0 {
```

Comment:
Feels like this should be moved at top of function, because in that case, we only need `validFor` and not `expiresAt`.

Reply:
Moved